lib/net/ldap/auth_adapters/simple.rb

```diff
@@ -0,0 +1,36 @@
+require 'net/ldap/auth_adapter'
+
+module Net
+  class LDAP
+    module AuthAdapters
+      class Simple < AuthAdapter
+        def bind(auth)
+          user, psw = if auth[:method] == :simple
+                        [auth[:username] || auth[:dn], auth[:password]]
+                      else
+                        ["", ""]
+                      end
+
+          raise Net::LDAP::BindingInformationInvalidError, "Invalid binding information" unless (user && psw)
+
+          message_id = @connection.next_msgid
+          request    = [
+            Net::LDAP::Connection::LdapVersion.to_ber, user.to_ber,
+            psw.to_ber_contextspecific(0)
+          ].to_ber_appsequence(Net::LDAP::PDU::BindRequest)
+
+          @connection.send(:write, request, nil, message_id)
```

Comment:
I agree, but this is the existing implementation, so let's change this in a separate PR.